coin_chk_libhdr.m4

```diff
@@ -0,0 +1,432 @@
+# of the link check ('link'). [dfltaction] ($2) is the parameter to
+# be analysed. [dfltuse] ($3) specifies the default for usage, in case
+# [dfltaction] doesn't specify it. [dfltlink] ($4) covers the default for
+# the link check, in case [dfltaction] doesn't specify it.
+
+# It's critical that the macro produce a result with no leading or trailing
+# spaces! Be careful if you modify it. In normal use, you want this macro to
+# expand immediately, so don't quote it.
+
+AC_DEFUN([AC_COIN_LIBHDR_DFLT_HACK],
+[m4_case([$2],
+   [yessep],[m4_if([$1],[link],[sep],[yes])],
+   [yestog],[m4_if([$1],[link],[tog],[yes])],
+   [nosep],[m4_if([$1],[link],[sep],[no])],
+   [notog],[m4_if([$1],[link],[tog],[no])],
+   [yes],[m4_if([$1],[link],[$4],[yes])],
+   [no],[m4_if([$1],[link],[$4],[no])],
+   [sep],[m4_if([$1],[link],[sep],[$3])],
+   [tog],[m4_if([$1],[link],[tog],[$3])],
+   [m4_if([$1],[link],[$4],[$3])])])dnl 	# COIN_LIBHDR_DFLT_HACK
+  
+
+
+###########################################################################
+#                         FIND_PRIM_LIBHDR                                #
+###########################################################################
+
+# COIN_FIND_PRIM_LIBHDR([prim],[lflgs],[cflgs],[dflgs],
+#                       [function-body],[includes],
+#                       [dfltaction],[cmdlineopts])
+
+# Determine whether we can use primary library prim ($1) and assemble
+# information on the required linker flags (prim_lflags), compiler flags
+# (prim_cflags), and data directories (prim_data) as specified by cmdlineopts
+# ($8).
+
+# A compile check will be performed if [includes] ($6) is specified.  A link
+# check will be performed if [function-body] ($5) is specified. The default is
+# to use a program composed by concatenating [includes] and [function-body]
+# with link flags specified by [lflgs] ($2), and compiler flags specified by
+# [cflgs] ($3). If [dfltaction] ($7) ends in 'sep', the link check will be
+# performed without the header (see LIBHDR_DFLT_HACK).
+
+# There's really no reasonable assumptions that can be made about how a data
+# directory is specified, so there's no attempt to test for existence.
+
+# [cmdlineopts] ($8) specifies the set of configure command line options
+# processed: 'nodata' processes --with-prim, --with-prim-lflags,
+# and --with-prim-cflags; 'dataonly' processes only --with-prim and
+# --with-prim-data; anything else ('all' works well) processes all four
+# command line options. Shell code produced by the macro is tailored based
+# on [cmdlineopts]. `nodata' is the default.
+
+# --with-prim is interpreted as follows:
+#   * --with-prim=no is equivalent to --without-prim
+#     prim status is set to skipping
+#   * --with-prim or --with-prim=yes is equivalent to
+#       --with-prim-lflags=-lprim
+#       --with-prim-data=/usr/local/share/prim
+#     if the user doesn't override lflags & data from the command line
+#     prim status is set to requested
```

Comment:
I wonder whether `--with-prim` without argument or `--with-prim=yes` should just give an error.

When the system does some guesses like here, it tries to be helpful, but it can make it difficult to understand what is going on eventually. I find it better if the user has to be explicit.

So the help should suggest `--without-prim`, `--with-prim-lflags`, `--with-prim-cflags`, `-with-prim-data` and we can still interpret `--with-prim=xyz` as `--with-prim-lflags=xyz` for a bit of backward compatibility.

Reply:
Stumbled over this point while working on `CHK_LAPACK`. I'm inclined to agree. `with-prim` should be a yes/no thing. `with-prim=no` is absolute, any values for `with-prim-lflags` or `with-prim-cflags` are ignored. `with-prim=yes` says use the values of `with-prim-lflags` and `with-prim-cflags`, whatever they might be (including null) with no guessing. If this fails (and it likely wil), let it fail. We can then suggest supplying correct values for lflags & cflags.

The way the macros are written, if the developer has set the default for a package to `no`, an explicit `with-prim=yes` is required. Without it, values for `with-prim-lflags` and `with-prim-cflags` are ignored. This might not be the most intuitive approach. I can see an argument that says supplying values for `with-prim-lflags` and `with-prim-cflags` should imply `with-prim=yes`. Can be done, just makes the command line processing logic a bit more complicated.

Reply:
Would be nice if specifying one of the `--with-prim-xyz` automatically enables the package and does not require an extra `--with-prim`.

Reply:
I was afraid you'd say that  :-)    I'll work on it.